contracts/src/pkg/XApp.sol

```diff
@@ -28,8 +45,10 @@ contract XApp {
         delete xmsg;
     }
 
-    constructor(address _omni) {
-        omni = IOmniPortal(_omni);
+    // TODO: write XAppUpgradeable that follows initialize pattern
+    constructor(address omni_, uint8 defaultConfLevel_) {
+        _setOmniPortal(omni_);
+        _setDefaultConfLevel(defaultConfLevel_);
     }
```

Comment:
Adding state to XApp will require an Upgradeable option. Or just to not have a contructor and leave `_setOmniPortal` and `_setDefaultConfLevel` to users.

Additionally, I changed omni from immutable to storage, because we should leave the option for xapps to change their portal address, to allow us the option to deploy now portals and migrate xapps over.

Reply:
Well I think it needs to be state. Adding another contract to handle something like storing the portal addr or default conf level of xapp deployed separately would make things more complicated